### PR TITLE
[BAD-569]: Unable to load for editing ktr with HBase Input/Output steps if KDC host is unavailable

### PR DESCRIPTION
**Pull request by TatsianaKasiankova**
This issue was introduced with changes from 6.0 to 6.1 related with OSGIing Hbase: BACKLOG-3896
see commit: ba87357
More detailed the cause of the issue and investigations are described in the JIRA: http://jira.pentaho.com/browse/BAD-569?focusedCommentId=281763&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-281763

The fix consists in don't to use HBase service for HBase steps on design time (in meta and dialog) but create HBaseValueMetaInterfaceFactory, ColumnFilterFactory and MappingFactory locally as it was before.
Also it has been added unit tests, fixed minor checkstyle issues and using deprecated method Const.isEmpty.

### CODE REVIEW

**Comment by TatsianaKasiankova** [comment]
@pentaho-nbaker Hi, Nick.
The implementation of the fix was discussed in the letter with you (please see the letter with summary: ).
Please take a look on this PR. 
About **unit test coverage changes** - It seems that an error occurs with calculating unit test coverage.
**Because:**
E.g. I've added two new test classes - but they are lost in test coverage statistic:
  kettle-plugins/hbase/src/test/java/org/pentaho/big/data/kettle/plugins/hbase/HBaseDataLoadSaveUtilTest.java
  kettle-plugins/hbase/src/test/java/org/pentaho/big/data/kettle/plugins/hbase/HBaseServiceLocalImpTest.java

For several classes I didn't changes at all (e.g. org.pentaho.big.data.kettle.plugins.hbase.NamedClusterLoadSaveUtil and etc.) , but the test coverage statistic show changes.
Also test coverage statistic show changes for he classes that have no unit tests at all - e. g. org.pentaho.big.data.kettle.plugins.hbase.FilterDefinition, 
org.pentaho.big.data.kettle.plugins.hbase.MappingDefinition and etc.

**Comment by TatsianaKasiankova** [comment]
@pentaho-nbaker Hi, Nick.
The implementation of the fix was discussed in the letter with you (please see the letter with summary: BAD-569: It needs advise to choose the way to resolve this problem).
Please take a look on this PR.
Unit test coverage change is Ok.
Tested locally.

File: assemblies/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -61,6 +61,8 @@
     <bundle>mvn:pentaho/pentaho-big-data-kettle-plugins-common-job/${project.version}</bundle>
     <bundle>mvn:pentaho/pentaho-big-data-kettle-plugins-common-named-cluster-bridge/${project.version}</bundle>
     <bundle>mvn:pentaho/pentaho-big-data-kettle-plugins-guiTestActionHandlers/${project.version}</bundle>
+    <bundle>wrap:mvn:pentaho/pentaho-hadoop-shims-api/${project.version}</bundle>
```
Comment (VasilinaTerehova) [inline assemblies/features/src/main/resources/features.xml]: api will be now loaded from both big-data-plugin lib folder and from osgi, right? we have legacy-plugin jar which use pentaho-hadoop-shims-api loaded not from osgi, and calls to api library classes from it, according dependency in maven https://github.com/pentaho/big-data-plugin/blob/master/legacy/pom.xml#L51, while other bundles will use api classes loaded from another classloader - osgi, which can come to classcastexceptions, full tests of impersonated shims and ordinary shim are highly required

File: kettle-plugins/hbase/src/main/java/org/pentaho/big/data/kettle/plugins/hbase/HBaseServiceLocalImp.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.pentaho.bigdata.api.hbase.mapping.MappingFactory;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.core.variables.VariableSpace;
+
+import com.pentaho.big.data.bundles.impl.shim.hbase.ByteConversionUtilImpl;
+import com.pentaho.big.data.bundles.impl.shim.hbase.mapping.ColumnFilterFactoryImpl;
+import com.pentaho.big.data.bundles.impl.shim.hbase.mapping.MappingFactoryImpl;
+import com.pentaho.big.data.bundles.impl.shim.hbase.meta.HBaseValueMetaInterfaceFactoryImpl;
+
+public class HBaseServiceLocalImp implements HBaseService {
```
Comment (TatsianaKasiankova) [inline kettle-plugins/hbase/src/main/java/org/pentaho/big/data/kettle/plugins/hbase/HBaseServiceLocalImp.java]: Yes

Comment (VasilinaTerehova) [inline kettle-plugins/hbase/src/main/java/org/pentaho/big/data/kettle/plugins/hbase/HBaseServiceLocalImp.java]: now - HBaseServiceLocalImp, need to be updated

**Comment by TatsianaKasiankova** [comment]
Going to re implement to avoid using methods in pentaho/pentaho-big-data-impl-shim-hbase for the work that is not related to shims.

**Pull request closed without merging**